x-pack/plugins/security_solution/public/resolver/data_access_layer/mocks/emptiable_no_ancestors_two_children.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ResolverRelatedEvents,
+  ResolverTree,
+  ResolverEntityIndex,
+} from '../../../../common/endpoint/types';
+import { mockEndpointEvent } from '../../mocks/endpoint_event';
+import {
+  mockTreeWithNoAncestorsAnd2Children,
+  mockTreeWithNoProcessEvents,
+} from '../../mocks/resolver_tree';
+import { DataAccessLayer } from '../../types';
+
+interface EmptiableRequests {
+  indexPatterns?: boolean;
+  relatedEvents?: boolean;
+  resolverTree?: boolean;
+  entities?: boolean;
+}
+
+interface Metadata {
+  /**
+   * The `_id` of the document being analyzed.
+   */
+  databaseDocumentID: string;
+  /**
+   * A record of entityIDs to be used in tests assertions.
+   */
+  entityIDs: {
+    /**
+     * The entityID of the node related to the document being analyzed.
+     */
+    origin: 'origin';
+    /**
+     * The entityID of the first child of the origin.
+     */
+    firstChild: 'firstChild';
+    /**
+     * The entityID of the second child of the origin.
+     */
+    secondChild: 'secondChild';
+  };
+}
+
+/**
+ * A simple mock dataAccessLayer that allows you to configure which requests return an empty set of data.
```

Comment:
:information_source: good idea. These empty kinds seem to give us a lot of trouble, We should use this a lot in tests.